stacktrace.cc

```diff
@@ -69,24 +69,23 @@ void release_backtrace(void* addr)
 	pr_dbg("  release_backtrace(%p)\n", addr);
 
 	const auto& addrit = addrmap.find(addr);
-	if (addrit == addrmap.end())
+	if (unlikely(addrit == addrmap.end()))
 		return;
 
 	object_info_t& object_info = addrit->second;
 	stack_trace_t& stack_trace = object_info.stack_trace;
 
 	const auto& stackit = stackmap.find(stack_trace);
-	if (stackit == stackmap.end())
+	if (unlikely(stackit == stackmap.end()))
 		return;
 
 	stack_info_t& stack_info = stackit->second;
 	stack_info.total_size -= object_info.size;
 	stack_info.count--;
 	if (stack_info.count == 0) {
-		const auto& it = stackmap.find(stack_trace);
-		if (it != stackmap.end())
-			stackmap.erase(it);
+		stackmap.erase(stackit);
 	}
+	addrmap.erase(addrit);
```

Comment:
I would like to suggest this part as follows.
```c
	if (stack_info.count == 0) {
		// The stackmap for the given stacktrace is no longer needed.
		stackmap.erase(stackit);
	}

	// The given address is released so remove it from addrmap.
	addrmap.erase(addrit);
```

Reply:
I changed the patch as you recommended